rpcserver.go

```diff
@@ -2545,6 +2527,43 @@ func (r *rpcServer) SendToRoute(stream lnrpc.Lightning_SendToRouteServer) error
 	})
 }
 
+// unmarshallSendToRouteRequest unmarshalls an rpc sendtoroute request
+func unmarshallSendToRouteRequest(req *lnrpc.SendToRouteRequest,
+	graph *channeldb.ChannelGraph) (*rpcPaymentRequest, error) {
+
+	switch {
+	case len(req.Routes) == 0 && req.Route == nil:
+		return nil, fmt.Errorf("unable to send, no routes provided")
+	case len(req.Routes) > 0 && req.Route != nil:
+		return nil, fmt.Errorf("cannot use both route and routes field")
+	}
+
+	var routes []*routing.Route
+	if len(req.Routes) > 0 {
```

Comment:
Why even distinguish these two cases? If the case of the `else` statement, it completes after a single loop iteration.

Reply:
I chose not to restrict the `Routes` field to a single element, but to add a new non-list field `Route`. Therefore the else construct here.

The intention behind this is to make one change to the interface and not two (first restrict to a single element and then later make it a non-list field).